### PR TITLE
Allow non-static library for `moduleLB`

### LINKED ISSUE #3085: False negative on EveryLB support test
```
CMake Error at cmake/FindCharm.cmake:63 (message):
  Could not find charm module EveryLB.Make sure you have built the LIBS
  target when building charm++
Call Stack (most recent call first):
  cmake/SetupCharm.cmake:4 (find_package)
  CMakeLists.txt:70 (include)
```
But if I remove the CMake check the code compiles and runs.

Probably the issue is that it only checks for a static library.

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Fixes #3085 (I think -- @wthrowe can you confirm?)

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: cmake/FindCharm.cmake [hunk 1]
```diff
@@ -54,13 +54,13 @@ find_library(CHARM_LIBCK
     HINTS ${CHARM_ROOT} ENV CHARM_ROOT
 )
 find_library(CHARM_EVERYLB
-    NAMES libmoduleEveryLB.a
+    NAMES moduleEveryLB
```
Comment (nilsvu) [inline cmake/FindCharm.cmake]: Oh we should have caught this when reviewing #3077, sorry.

I think you should also add `NO_DEFAULT_PATH`, replace `CHARM_ROOT` with `CHARM_LIBRARIES` (deleting the `ENV` part) and move the `find_library` call down below the definition of `CHARM_LIBRARIES`. My intention is that the library is only searched in the libs directory that is actually linked. Docs: https://cmake.org/cmake/help/latest/command/find_library.html

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: cmake/FindCharm.cmake (hunk 1 above)
Comment (moxcodes) [reply cmake/FindCharm.cmake]: makes sense; squashed in the changes.

File: cmake/FindCharm.cmake [hunk 2]
```diff
@@ -53,19 +53,20 @@ find_library(CHARM_LIBCK
...
+get_filename_component(CHARM_LIBRARIES ${CHARM_LIBCK} DIRECTORY)
+
 find_library(CHARM_EVERYLB
-    NAMES libmoduleEveryLB.a
-    PATH_SUFFIXES lib64 lib
-    HINTS ${CHARM_ROOT} ENV CHARM_ROOT
-)
+  NAMES moduleEveryLB
+  PATH_SUFFIXES lib64 lib
+  HINTS ${CHARM_ROOT} CHARM_LIBRARIES
```
Comment (nilsvu) [inline cmake/FindCharm.cmake]:
```suggestion
  HINTS ${CHARM_LIBRARIES}
  NO_DEFAULT_PATH
```

(currently the CHARM_LIBRARIES as taken as a literal string I think)

Comment (moxcodes) [reply cmake/FindCharm.cmake]: ah, yes, :facepalm: my mistake

Comment (moxcodes) [reply cmake/FindCharm.cmake]: Squashed. thanks!

**Review by nilsvu — DISMISSED** [review]
LGTM :)

File: cmake/FindCharm.cmake [hunk 3]
```diff
@@ -53,19 +53,20 @@ find_library(CHARM_LIBCK
...
+get_filename_component(CHARM_LIBRARIES ${CHARM_LIBCK} DIRECTORY)
+
 find_library(CHARM_EVERYLB
-    NAMES libmoduleEveryLB.a
-    PATH_SUFFIXES lib64 lib
-    HINTS ${CHARM_ROOT} ENV CHARM_ROOT
-)
+  NAMES moduleEveryLB
+  HINTS ${CHARM_LIBRARIES}
+  NO_DEFUALT_PATH
```
Comment (nilsvu) [inline cmake/FindCharm.cmake]: Oops `NO_DEFAULT_PATH`

**Review by wthrowe — DISMISSED** [review]
Seems to fix things.

File: cmake/FindCharm.cmake (hunk 3 above)
Comment (moxcodes) [reply cmake/FindCharm.cmake]: boy I wish cmake could have better syntax checking...

**Comment by moxcodes** [comment]
squashed defualt->default

**Review by wthrowe — APPROVED** [review]
Still works.

**Review by nilsvu — APPROVED** [review]

**Comment by nilsvu** [comment]
Merging now to restore builds on systems where charm is built with shared libs.

**Pull request merged**